src/libstd/collections/hash/set.rs

```diff
@@ -764,18 +764,27 @@ pub struct Iter<'a, K: 'a> {
     iter: Keys<'a, K, ()>
 }
 
+unsafe impl<'a, K: Sync> Sync for Iter<'a, K> {}
+unsafe impl<'a, K: Sync> Send for Iter<'a, K> {}
```

Comment:
Er actually, all of the modifications in this file should not be necessary as they're built on top of the map iterators